Modules/CalcPerform.lua

```diff
@@ -1595,7 +1886,603 @@ function calcs.perform(env)
 			end
 		end
 	end
-	
+
+	-- Process Triggered Skill and Set Trigger Conditions
```

Comment:
~~There seems to be a decent amount of duplication in this section, other than the item names and some flags that apply.  If there isn't a clean way to put some of it into a function, that's fine, but it's adding a lot of mental overhead when making changes here.~~

Actually, on second thought, these are pretty specific to each item in how they trigger, so maybe it's fine to keep them with distinct logic.